src/main/java/duke/Chatbot.java

```diff
@@ -0,0 +1,121 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.Scanner;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.task.TaskType;
+import duke.task.ToDo;
+
+
+/**
+ * A duke.Duke chatbot that provides todo-list function for users.
+ */
+public class Chatbot {
+    private final Storage storage;
+    private final TaskList taskList;
+    private final Parser parser;
+
+    /**
+     * Constructs the chat bot.
+     */
+    public Chatbot() {
+        storage = new Storage("./data.txt");
+        this.taskList = storage.readFile();
+        parser = new Parser();
+    }
+
+
+    /**
+     * Executes the chat bot. Keep looping and asking for user input.
+     * The function will be terminated once hit "bye".
+     */
+    public void execute() {
+        Scanner sc = new Scanner(System.in);
+        String input = sc.nextLine();
+
+        while (!input.toLowerCase().equals("bye")) {
+            System.out.println(getDukeReply(input));
+            input = sc.nextLine();
+        }
+
+
+    }
+
+    public String getDukeReply(String input) {
+        String[] taskTimeSplit;
+        Task newTask;
+        try {
+            String[] taskTypeSplit = input.split(" ");
```

Comment:
Perhaps calling it inputs could be clearer since you are splitting the input.

Reply:
Perhaps declaring what each index of the taskTypeSplit is could be clearer for readers.